src/coreclr/jit/llvm.cpp

```diff
@@ -235,8 +283,65 @@ Value* genTreeAsLlvmType(GenTree* tree, Type* type)
     failFunctionCompilation();
 }
 
-llvm::Value* buildUserFuncCall(GenTreeCall* gtCall, llvm::IRBuilder<>& builder)
+int getTotalParameterOffset(CORINFO_SIG_INFO &sigInfo)
+{
+    unsigned int offset = 0;
+    // TODO: we are not accepting any of these arg types in the clrjit compilation at present.  The only thing on the shadow stack can be the method's "this"
+    // for (int i = 0; i < sigInfo.numArgs; i++)
+    //{
+        //if (!CanStoreVariableOnStack(_signature[i]))
+        //{
+        //    offset = PadNextOffset(_signature[i], offset);
+        //}
+    //}
+    if (sigInfo.hasThis())
+    {
+        // If this is a struct, then it's a pointer on the stack
+        //if (_thisType.IsValueType)
+        //{
+        //    offset = PadNextOffset(_thisType.MakeByRefType(), offset);
+        //}
+        //else
+        //{
+        //    offset = PadNextOffset(_thisType, offset);
+        //}
+        // TODO: not as correct as the above, but don't know how to get all the field alignment values needed to implement the
+        // equivalent here.  How to get InstanceFieldSize, InstanceFieldAlignment, ComputePackingSize
+        offset = _pointerSize;
+    }
+
+    return AlignUp(offset, _pointerSize);
+}
+
+int getTotalLocalOffset(CORINFO_SIG_INFO& sigInfo)
+{
+    // TODO: need to store some locals on ths shadow stack, either when there are exception blocks, or they are, or have, GC pointers (so the conservative GC knows they are live)
+    // For now we don't have any so simply:
+    return 0;
+}
+
+llvm::Value* getShadowStackOffest(Value* shadowStack, unsigned int offset)
+{
+    if (offset == 0)
+        return shadowStack;
+
+    return _builder->CreateGEP(shadowStack, _builder->getInt32(_pointerSize));
+}
+
+bool isThisArg(GenTreeCall* gtCall, GenTree* operand)
 {
+    // TODO: is the this arg always a late arg?
```

Comment:
I think you can make this logic not dependent on the lateness of `this`:
```c++
if (gtCall->gtCallThisArg == nullptr)
{
    return false;
}
   
return _compiler->gtGetThisArg(gtCall) == operand;
```
Also nit: `gtCall` -> `call`. The purpose of the `gt`/`fg`/`gen`/etc is the namespacing of sorts for when everything is a `Compiler` member. I don't think there is need to use the prefixes for locals.